lib/reply.js

```diff
@@ -377,8 +377,8 @@ function preserializeHookEnd (err, request, reply, payload) {
 }
 
 function onSendHook (reply, payload) {
-  reply[kReplySent] = true
   if (reply.context.onSend !== null) {
+    reply[kReplySent] = true
```

Comment:
I think it should be completely removed as this value will be set inside `onSendHookRunner` and `onSendEnd`

Reply:
If I remove it completely, The following tests fail:
1. test/hooks-async.test.js 5 failed
2. test/hooks.test.js 1 failed